packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -149,124 +194,132 @@ export class TokenRatesController extends StaticIntervalPollingControllerV1<
+   * @param options.currentTicker - The ticker for the current network.
+   * @param options.currentAddress - The current selected address.
+   * @param options.disabled - Boolean to track if network requests are blocked
+   * @param options.tokenPricesService - An object in charge of retrieving token price
+   * @param options.messenger - The controller messaging system
+   * @param options.state - Initial state to set on this controller
    */
-  constructor(
-    {
-      interval = 3 * 60 * 1000,
-      threshold = 6 * 60 * 60 * 1000,
-      getNetworkClientById,
-      chainId: initialChainId,
-      ticker: initialTicker,
-      selectedAddress: initialSelectedAddress,
-      onPreferencesStateChange,
-      onTokensStateChange,
-      onNetworkStateChange,
-      tokenPricesService,
-    }: {
-      interval?: number;
-      threshold?: number;
-      getNetworkClientById: NetworkController['getNetworkClientById'];
-      chainId: Hex;
-      ticker: string;
-      selectedAddress: string;
-      onPreferencesStateChange: (
-        listener: (preferencesState: PreferencesState) => void,
-      ) => void;
-      onTokensStateChange: (
-        listener: (tokensState: TokensState) => void,
-      ) => void;
-      onNetworkStateChange: (
-        listener: (networkState: NetworkState) => void,
-      ) => void;
-      tokenPricesService: AbstractTokenPricesService;
-    },
-    config?: Partial<TokenRatesConfig>,
-    state?: Partial<TokenRatesState>,
-  ) {
-    super(config, state);
-    this.defaultConfig = {
-      interval,
-      threshold,
-      disabled: false,
-      nativeCurrency: initialTicker,
-      chainId: initialChainId,
-      selectedAddress: initialSelectedAddress,
-      allTokens: {}, // TODO: initialize these correctly, maybe as part of BaseControllerV2 migration
-      allDetectedTokens: {},
-    };
+  constructor({
+    interval = DEFAULT_INTERVAL,
+    currentChainId,
+    currentTicker,
+    currentAddress,
+    disabled = false,
+    tokenPricesService,
+    messenger,
+    state,
+  }: {
+    interval?: number;
+    currentChainId: Hex;
+    currentTicker: string;
+    currentAddress: string;
+    disabled?: boolean;
+    tokenPricesService: AbstractTokenPricesService;
+    messenger: TokenRatesControllerMessenger;
+    state?: Partial<TokenRatesControllerState>;
+  }) {
+    super({
+      name: controllerName,
+      messenger,
+      state: { ...getDefaultTokenRatesControllerState(), ...state },
+      metadata,
+    });
 
-    this.defaultState = {
-      contractExchangeRates: {},
-      contractExchangeRatesByChainId: {},
-    };
-    this.initialize();
     this.setIntervalLength(interval);
-    this.getNetworkClientById = getNetworkClientById;
     this.#tokenPricesService = tokenPricesService;
+    this.#disabled = disabled;
+    this.#interval = interval;
+    this.#chainId = currentChainId;
+    this.#ticker = currentTicker;
+    this.#selectedAddress = currentAddress;
```

Comment:
Having the selected address as a class variable here means that we also need to keep this in sync. Perhaps it would be better to always take the fresh value from `AccountsController`?